plugins/content/createmenulink/createmenulink.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Content.createmenulink
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\Registry\Registry;
+
+JLoader::import('joomla.application.component.model');
+
+/**
+ * Createmenulink Plugin
+ *
+ * @since  3.6
+ */
+class PlgContentCreatemenulink extends JPlugin
+{
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 * @since  3.6
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Runs on content preparation
+	 *
+	 * @param   string  $context  The context for the data
+	 * @param   object  $data     An object containing the data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentPrepareData($context, $data)
+	{
+		if (!in_array($context, array('com_content.article','com_contact.contact')))
+		{
+			return true;
+		}
+
+		$app = JFactory::getApplication();
+
+		if ($app->isAdmin())
+		{
+			list($component, $view) = explode('.', $context);
+
+			$menu      = JFactory::getApplication()->getMenu('site');
+			$menuItems = $menu->getItems(
+							'link',
+							'index.php?option=' . $component . '&view=' . $view . '&id=' . $data->id
+						);
+
+			if (!empty($menuItems))
+			{
+				$data->menuid    = $menuItems[0]->id;
+				$data->menutitle = $menuItems[0]->title;
+				$data->menualias = $menuItems[0]->alias;
+				$data->menutype  = $menuItems[0]->menutype;
+				$data->parent_id = $menuItems[0]->parent_id;
+			}
+
+			if (empty($menuItems))
+			{
+				JHtml::_('jquery.framework', false);
+				$document = JFactory::getDocument();
+				$document->addScriptDeclaration('
+				(function ($) {
```

Comment:
I am still for a own JavaScript file in the media folder here.. You can then load it with JHtml::_('script', 'media/path/to/file/js' ..

We have enough ugly inline JavaScript in Joomla.

P.S.: You are sure you only want it when you don't have a menu item? Shouldn't the menu item title update when the title changes? People will think it changes everytime and not only when they type it in the title first time.